module5/lessons/Week1/Logging.md

```diff
@@ -0,0 +1,142 @@
+
+## The Task
+
+Your task is to "Add logging to our MVC Movies application." You will be working with a partner and working in the traditional pair programming style, switching at least every 30 minutes between driver and navigator.
+
+This is a very open-ended task!
+
+Here are two guide posts:
+* You should log both to a file and to the console
+
+* Everyone should use the logging library [Serilog](https://serilog.net/).  It's highly recommended in the .NET community and while there are many options out there it will be easier to help each other debug if we stick to the same tech stack. There is a built in logger in .NET you can also look into but ["ASP.NET Core doesn't include a logging provider for writing logs to files."](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-7.0#built-in-logging-providers) so for that goal we need another tool.
+
+If other questions come up as you are working, post them in the codehelp Slack channel!
+
+**✅Deliverable✅:** With your partner write up the steps required to log to a file and the console in a .NET application. Include screenshots of what the log file and console log should look like when it's working. A teammate should be able to follow your guide and get an initial message to log. Include any bugs/problems you ran into during set-up so that your teammate can avoid making the same mistake. Send these instructions in a slack message to your instructor.
+<aside class="instructor-notes" markdown="1">
+    <p><strong>Instructor Note</strong><br>These getting started instructions are all it takes, it's only about 3 steps! https://github.com/serilog/serilog/wiki/Getting-Started </p>
+</aside>
+
+Once you complete this deliverable, dive deeper into the world of .NET logging! You want to build the highest quality logging you can in the time block provided. 
+
+<aside class="instructor-notes" markdown="1">
+    <p><strong>Instructor Note</strong><br>I think it would helpful to put a goal time for when students finish this deliverable. I'm thinking about half way through the alloted time for this lesson. To set the expectation that this is just the bare bones, they should definitely keep improving their logging from here. I'd also probably share lal of the guides with the class once ever group has finished.</p>
+</aside>
+
+**At some point your instructor will call the class back together to talk through some questions as a group and discuss tradeoffs you have run into.**
+
+<aside class="instructor-notes" markdown="1">
+    <p><strong>Instructor Note</strong><br>
+    Here are the questions:
+    * When should we use a console.log and when should we use logging?
+    * Who is using Seralog and who is using the built-in logger?
+    * What's one other tradeoff you and your partner have run into so far? What options did you consider? (then group discussion about what others think on this topic)
+    </p>
+</aside>
+
+## Tiny Next Goal

```

Comment:
This is something I totally just made up because I wanted some structure to help students break down a big problem into little steps. If you know of any other structure here definitely open to it!!